src/EffectiveCSharp.Analyzers/PreferDeclarationInitializersToAssignmentStatementsCodeFixProvider.cs

```diff
@@ -0,0 +1,257 @@
+namespace EffectiveCSharp.Analyzers;
+
+/// <summary>
+/// A <see cref="CodeFixProvider"/> that provides a code fix for the <see cref="PreferDeclarationInitializersToAssignmentStatementsAnalyzer"/>.
+/// </summary>
+/// <seealso cref="Microsoft.CodeAnalysis.CodeFixes.CodeFixProvider" />
+[ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(PreferDeclarationInitializersToAssignmentStatementsCodeFixProvider))]
+[Shared]
+public class PreferDeclarationInitializersToAssignmentStatementsCodeFixProvider : CodeFixProvider
+{
+    private static readonly IEqualityComparer<IFieldSymbol> _fieldSymbolNameComparer = new FieldSymbolNameComparer();
+
+    /// <inheritdoc />
+    public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(
+        DiagnosticIds.PreferDeclarationInitializersToAssignmentStatement,
+        DiagnosticIds.PreferDeclarationInitializersExceptNullOrZero,
+        DiagnosticIds.PreferDeclarationInitializersExceptWhenVaryingInitializations,
+        DiagnosticIds.PreferDeclarationInitializersWhenNoInitializationPresent);
+
+    /// <inheritdoc />
+    public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+    {
+        SyntaxNode? root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+        SemanticModel? semanticModel = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+
+        if (root is null || semanticModel is null)
+        {
+            return;
+        }
+
+        Dictionary<IFieldSymbol, IList<ExpressionStatementSyntax>> initializationsToFixPerField = new Dictionary<IFieldSymbol, IList<ExpressionStatementSyntax>>(_fieldSymbolNameComparer);
```

Comment:
YOu should be able to shorten this up

```suggestion
        Dictionary<IFieldSymbol, IList<ExpressionStatementSyntax>> initializationsToFixPerField = new (_fieldSymbolNameComparer);
```

Reply:
I don't need this anymore. Before, I wrote the provider thinking it would show me all diagnostics for a given analyzer, but instead its per span.